lib/dns.js

```diff
@@ -189,6 +189,9 @@ exports.lookupService = function(host, port, callback) {
   if (cares.isIP(host) === 0)
     throw new TypeError('"host" argument needs to be a valid IP address');
 
+  if (typeof port !== 'number')
```

Comment:
I think it might be better to coerce and check the result with `isFinite()`. Allowing a port number as a string here would be consistent with other subsystems that allow such inputs (e.g. `http.request()`).